crates/sui-indexer/migrations_v2/2023-08-19-044035_tx_indices/up.sql

```diff
@@ -0,0 +1,19 @@
+CREATE TABLE tx_indices (
+    tx_sequence_number          BIGINT       PRIMARY KEY,
+    transaction_digest          bytea        NOT NULL,
+    input_objects               bytea[]      NOT NULL,
+    changed_objects             bytea[]      NOT NULL,
+    senders                     bytea[]      NOT NULL,
+    recipients                  bytea[]      NOT NULL,
+    packages                    bytea[]      NOT NULL,
+    package_modules             text[]       NOT NULL,
+    package_module_functions    text[]       NOT NULL
+);
+
+CREATE INDEX tx_indices_input_objects ON tx_indices USING GIN(input_objects);
+CREATE INDEX tx_indices_chagned_objects ON tx_indices USING GIN(changed_objects);
+CREATE INDEX tx_indices_senders ON tx_indices USING GIN(senders);
+CREATE INDEX tx_indices_recipients ON tx_indices USING GIN(recipients);
+CREATE INDEX tx_indices_package ON tx_indices USING GIN(packages);
+CREATE INDEX tx_indices_package_module ON tx_indices USING GIN(package_modules);
+CREATE INDEX tx_indices_package_module_function ON tx_indices USING GIN(package_module_functions);
```

Comment:
Other queries in the GraphQL schema that there currently isn't support for here (from `TransactionBlockFilter`):

- `signAddress` (All addresses that signed the transaction).  Today, this includes the `sender` and potentially the sponsor.  Later, when we support multi-account signing, it could be arbitrarily many different addresses.
- `paidAddress` (The address that owns the gas coin). Either the sender or the sponsor, depending on whether it was sponsored or not.

Open question whether we want to support these or not (now, or ever).